lib/gears/filesystem/subdirectory_cache.lua

```diff
@@ -0,0 +1,125 @@
+---------------------------------------------------------------------------
+-- Check if some file exists inside a directory.
+--
```

Comment:
If it's a file (and subdir) cache, then I think calling this `directory_cache` would make more sense, no?